llvm/test/CodeGen/AIE/aie2/GlobalISel/combine-vinsert-sequence-prelegalizer.mir

```diff
@@ -328,3 +328,375 @@ body:             |
+    ; CHECK-NEXT: [[AIE_PAD_VECTOR_UNDEF:%[0-9]+]]:_(<32 x s16>) = G_AIE_PAD_VECTOR_UNDEF [[BUILD_VECTOR_TRUNC]](<16 x s16>)
+    ; CHECK-NEXT: [[BITCAST:%[0-9]+]]:_(<16 x s32>) = G_BITCAST [[AIE_PAD_VECTOR_UNDEF]](<32 x s16>)
+    ; CHECK-NEXT: PseudoRET implicit $lr, implicit [[BITCAST]](<16 x s32>)
+    %0:_(<16 x s16>) = G_INTRINSIC intrinsic(@llvm.aie2.v16int16)
+    %100:_(<8 x s32>) = G_BITCAST %0(<16 x s16>)
+    %1:_(s32) = G_CONSTANT i32 0
+    %2:_(s32) = G_CONSTANT i32 1
+    %3:_(s32) = G_CONSTANT i32 2
+    %4:_(s32) = G_CONSTANT i32 3
+    %5:_(s32) = G_CONSTANT i32 4
+    %6:_(s32) = G_CONSTANT i32 5
+    %7:_(s32) = G_CONSTANT i32 6
+    %8:_(s32) = G_CONSTANT i32 7
+    %9:_(s32) = G_CONSTANT i32 8
+    %10:_(s32) = G_CONSTANT i32 9
+    %11:_(s32) = G_CONSTANT i32 10
+    %12:_(s32) = G_CONSTANT i32 11
+    %13:_(s32) = G_CONSTANT i32 12
+    %14:_(s32) = G_CONSTANT i32 13
+    %15:_(s32) = G_CONSTANT i32 14
+    %16:_(s32) = G_CONSTANT i32 15
+    %101:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.set.I512.I256), %100(<8 x s32>), %1(s32)
+    %17:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.vinsert16.I512), %101(<16 x s32>), %1(s32), %16(s32)
+    %18:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.vinsert16.I512), %17(<16 x s32>), %2(s32), %14(s32)
+    %19:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.vinsert16.I512), %18(<16 x s32>), %3(s32), %12(s32)
+    %20:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.vinsert16.I512), %19(<16 x s32>), %4(s32), %10(s32)
+    %21:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.vinsert16.I512), %20(<16 x s32>), %5(s32), %8(s32)
+    %22:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.vinsert16.I512), %21(<16 x s32>), %6(s32), %6(s32)
+    %23:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.vinsert16.I512), %22(<16 x s32>), %7(s32), %4(s32)
+    %24:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.vinsert16.I512), %23(<16 x s32>), %8(s32), %2(s32)
+    %25:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.vinsert16.I512), %24(<16 x s32>), %9(s32), %1(s32)
+    %26:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.vinsert16.I512), %25(<16 x s32>), %10(s32), %3(s32)
+    %27:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.vinsert16.I512), %26(<16 x s32>), %11(s32), %5(s32)
+    %28:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.vinsert16.I512), %27(<16 x s32>), %12(s32), %7(s32)
+    %29:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.vinsert16.I512), %28(<16 x s32>), %13(s32), %9(s32)
+    %30:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.vinsert16.I512), %29(<16 x s32>), %14(s32), %11(s32)
+    %31:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.vinsert16.I512), %30(<16 x s32>), %15(s32), %13(s32)
+    %32:_(<16 x s32>) = G_INTRINSIC intrinsic(@llvm.aie2.vinsert16.I512), %31(<16 x s32>), %16(s32), %15(s32)
+    PseudoRET implicit $lr, implicit %32
+...
+
+---
+name:            vinsert16-I128_idx_miss
```

Comment:
Perhaps a very short description of why these cases do not yield a nice G_BUILD_VECTOR_TRUNC